### DISCUSSION TITLE
Presence of LaTeX Math breaks relative links in README.md

### DISCUSSION THREAD

**Discussion by mkXla2135** [community/community#61164] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

It appears that the relative links in `README.md` are not rendered correctly when math commands (via `$...$`) are present and the README file is rendered when visiting `https://github.com/<user>/<repo>` in the browser.

# Correct behavior
E.g., consider what happens when the `README.md` in the repository contains the following code:
```markdown
[link](./README.md) 
```
The relative link `./` gets rendered to `https://github.com/<user>/<repo>/blob/<branch>/`, resulting in a valid link `https://github.com/<user>/<repo>/blob/<branch>/README.md`, as described on [GitHub Docs](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#relative-links). As an example, I created a repository demonstrating this (correct) behavior [here](https://github.com/mkXla2135/test1).

# Incorrect behavior
Now, consider the following case: 
```markdown
[link](./README.md)  and $x$
```
When viewing the repository at `https://github.com/<user>/<repo>`, the relative link now is rendered to `https://github.com/<user>/README.md`, which, of course, gives 404 error. See an example repository [here](https://github.com/mkXla2135/test2).

When clicking the markdown file in the file list for viewing it separately (i.e., the browser opens `https://github.com/<user>/<repo>/blob/<branch>/README.md`), the links are generated correctly. The links are also generated correctly when opening `https://github.com/<user>/<repo>/tree/<branch>`. However, having this as a workaround is quite inconvenient when, e.g., the readme files describe the contents of the repository and users want to navigate to the necessary files directly when opening the repository.

**Comment by GhostOf0days** [comment]
As an additional comment, the LaTeX ReadMe's relative link works correctly on the mobile app, but breaks on the website.

**Reply by dvandyk** [reply]
This is a recent development. I had released a set of data files in May 2023 and relative links + LaTeX used to work at the time. The breaking has been introduced at some point between May 10th and July 18th.

**Comment by Zachary-Harrison** [comment]
I'm experiencing a similar issue when trying to render colored text via LaTeX. Took me ages to figure out, probably because it wasn't breaking in one of my projects. I did a little experimenting and I *think* that the relative links only break when they point to files, rather than directories. 

For an example of a successful relative link with LaTeX math mode, see [here](https://github.com/Zachary-Harrison/GIMP-Documentation#project-structure).
